src/client/Client.cc

```diff
@@ -12973,7 +12981,21 @@ int Client::ll_statfs(Inode *in, struct statvfs *stbuf, const UserPerm& perms)
   /* Since the only thing this does is wrap a call to statfs, and
      statfs takes a lock, it doesn't seem we have a need to split it
      out. */
-  return statfs(0, stbuf, perms);
+  return _statfs(in, stbuf, perms);
+}
+
+int Client::statfs(const char *path, struct statvfs *stbuf, const UserPerm& perms)
+{
+  walk_dentry_result wdr;
+  {
+    std::scoped_lock l(client_lock);
+    if (int rc = path_walk(cwd, filepath(path), &wdr, perms, {}); rc < 0) {
+      return rc;
+    }
+  }
+
+  auto in = wdr.target.get();
+  return _statfs(in, stbuf, perms);
```

Comment:
I think it's weird to not hold `client_lock` when calling the internal helper method. It allows for weird and unexpected races with interleaved calls into the Client. We should just restructure this so that `_statfs` gets a `std::unique_lock` argument with the lock held. (Fix `ll_statfs` above.)

Reply:
Done.

Note: I ended up using `client_lock.unlock()` and `client_lock.lock()` instead of adding an argument.